File: public/docs/_examples/structural-directives/dart/lib/structural_directives_component.dart
// #docplaster
// #docregion
import 'package:angular2/core.dart';

import 'heavy_loader_component.dart';
import 'unless_directive.dart';

@Component(
    selector: 'structural-directives',
    templateUrl: 'structural_directives_component.html',
    styles: const ['button { min-width: 100px; }'],
    directives: const [UnlessDirective, HeavyLoaderComponent])
class StructuralDirectivesComponent {
  List<String> heroes = ['Mr. Nice', 'Narco', 'Bombasto'];
  bool condition = true;
  bool isVisible = true;
  List<String> logs = [];
  String status = 'ready';

  get hero => heroes[0];
}

File: public/docs/_examples/structural-directives/ts/src/app/structural-directives.component.ts
// #docplaster
// #docregion
import { Component } from '@angular/core';

@Component({
  moduleId: module.id,
  selector: 'structural-directives',
  templateUrl: './structural-directives.component.html',
  styles: ['button { min-width: 100px; }']
})
export class StructuralDirectivesComponent {
  heroes = ['Mr. Nice', 'Narco', 'Bombasto'];
  hero = this.heroes[0];
  condition = true;
  isVisible = true;
  logs: string[] = [];
  status = 'ready';
}
// #enddocregion
